src/sqlfluff/utils/reflow/classes.py

```diff
@@ -0,0 +1,290 @@
+
+    It holds no configuration and is influenced by the blocks either
+    side.
+    """
+
+    def trailing_whitespace_fixes(self) -> List[LintFix]:
+        """Fix any trailing whitespace detected.
+
+        Trailing whitespace is unique in that it can be detected
+        solely within the reflow point. It is any whitespace
+        preceeding a newline.
+        """
+        # NOTE: We will need an end_of_file marker to do file ends.
+        reflow_logger.debug("TW: Checking %s", self)
+        if not self.class_types.intersection({"newline", "end_of_file"}):
+            return []
+        fixes = []
+        for idx, seg in enumerate(self.segments):
+            if not (
+                seg.is_type("newline", "end_of_file")
+                and idx > 0
+                and self.segments[idx - 1].is_type("whitespace")
+            ):
+                continue
+            # It's a trailing newline in the templated file. Check whether
+            # it is in the source.
+            trailing_whitespace = self.segments[idx - 1]
+            # Often, what looks like trailing whitespace in rendered SQL is
+            # actually a line like:
+            # "    {% for elem in elements %}\n"
+            # in which case the code is fine -- it's not trailing whitespace
+            # from a source code perspective.
+            if seg.pos_marker.templated_file:
+                next_raw_slice = (
+                    seg.pos_marker.templated_file.raw_slices_spanning_source_slice(
+                        slice(
+                            trailing_whitespace.pos_marker.source_slice.stop,
+                            trailing_whitespace.pos_marker.source_slice.stop,
+                        )
+                    )
+                )
+                # If the next slice is literal, that means it's regular code, so
+                # it's safe to delete the trailing whitespace. If it's anything
+                # else, it's template code, so don't delete the whitespace because
+                # it's not REALLY trailing whitespace in terms of the raw source
+                # code.
+                if next_raw_slice and next_raw_slice[0].slice_type != "literal":
+                    reflow_logger.debug(
+                        "    !! TW: Skipping apparent trailing whitespace "
+                        "followed by template"
+                    )
+                    continue
+
+            fixes.append(LintFix("delete", trailing_whitespace))
+            reflow_logger.debug("    !! TW: TRAILING WHITESPACE")
+        return fixes
+
+
+@dataclass
+class ReflowSequence:
+    """Class for keeping track of elements in a reflow operation.
+
+    It is assumed that there will be alternating blocks and points
+    (even if some points have no segments). This is validated on
+    construction.
+
+    We assume points on each end because this is the case with a file.
+    """
+
+    elements: Sequence[Union[ReflowBlock, ReflowPoint]]
```

Comment:
Since both `ReflowBlock` and `ReflowPoint` inherit from `_ReflowElement`, would it be reasonable to say:
```
Sequence[_ReflowElement]
```
?